src/main/scala/nl/knaw/dans/easy/ingest/EasyIngest.scala

```diff
@@ -199,34 +199,19 @@ object EasyIngest {
       request = (datastreamId, sdo.listFiles.find(_.getName == dsSpec.contentFile)) match {
         case ("EMD", Some(file)) =>
           // Note that this would change the ingested file's checksum, but it is only for the EMD datastream, which has no checksum
-          return executeAddRequestWithReplacement(request, file, PLACEHOLDER_FOR_DMO_ID, pidDictionary(sdo.getName))
+          request.content(replacePlaceHolder(file, PLACEHOLDER_FOR_DMO_ID, pidDictionary(sdo.getName)))
```

Comment:
Looks interesting! Vaguely remember hearing about it before... Never used it myself, but is seems like a good library. Also interesting that they use monadic style (`map`, `flatMap`, etc.). Is the same as in `Option`, `Try` and `List`.

My version was inspired by the [`using` from Rx](http://reactivex.io/RxJava/javadoc/rx/Observable.html#using(rx.functions.Func0,%20rx.functions.Func1,%20rx.functions.Action1)), which does the same, but for the `Observable`...

@jo-pol is trying this out now!